label_maker/label.py

```diff
@@ -66,99 +69,124 @@ def make_labels(dest_folder, zoom, country, classes, ml_type, bounding_box, spar
+            global tile_results
+            if tile_results.get(index) is None:
+                tile_results[index] = empty_label
+
+        # Print a summary of the labels
+        _tile_results_summary(ml_type, classes)
+
+        # If the --sparse flag is provided, limit the total background tiles to write
+        if sparse:
+            pos_examples, neg_examples = [], []
+            for k in tile_results.keys():
+                # if we don't match any class, this is a negative example
+                if not sum([class_match(ml_type, tile_results[k], i + 1) for i, c in enumerate(classes)]):
+                    neg_examples.append(k)
+                else:
+                    pos_examples.append(k)
+
+            # Choose random subset of negative examples
+            n_neg_ex = int(kwargs['background_ratio'] * len(pos_examples))
+            neg_examples = np.random.choice(neg_examples, n_neg_ex, replace=False).tolist()
+
+            tile_results = {k: tile_results.get(k) for k in pos_examples + neg_examples}
+            print('Using sparse mode; subselected {} background tiles'.format(n_neg_ex))
+
+        # write out labels as numpy arrays
+        labels_file = op.join(dest_folder, 'labels.npz')
+        # print('Writing out labels to {}'.format(labels_file))
+        np.savez(labels_file, **tile_results)
+
+        # write out labels as GeoJSON or PNG
+        if ml_type == 'classification':
+            features = []
+            label_area = []
+            label_bool = []
+            i = 0
+            for tile, label in tile_results.items():
+                label_bool.append([int(bool(l)) for l in label])
+                label_area.append([float(l) for l in label])
+                if 'label_bool2' in locals() and 'label_area2' in locals():
+                    label_bool[i] = [int(bool(label_bool[i] + label_bool2[i])) for label_bool[i], label_bool2[i] in zip(label_bool[i], label_bool2[i])]
+                    label_area[i] = [label_area[i] + label_area2[i] for label_area[i], label_area2[i] in zip(label_area[i], label_area2[i])]
+                # if there are no classes, activate the background
+                if ctr == country[-1]:
+                    if all(v == 0 for v in label_bool[i]):
+                        label_bool[i][0] = 1
+                    feat = feature(Tile(*[int(t) for t in tile.split('-')]))
+                    features.append(Feature(geometry=feat['geometry'],
+                                            properties=dict(feat_id=str(tile),
+                                                            label=label_bool[i],
+                                                            label_area=label_area[i])))
+                i += 1
+            label_bool2 = label_bool
+            label_area2 = label_area
+            if ctr == country[-1]:
+                json.dump(fc(features), open(op.join(dest_folder, f'classification_{zoom}.geojson'), 'w'))
```

Comment:
added `zoom`  to file name to identify the zoom level at which labelled output was generated.